code/workspaces/infrastructure-api/src/auth/permissionMiddleware.js

```diff
@@ -32,42 +32,31 @@ export function projectPermissionWrapper(permissionSuffix) {
   };
 }
 
-export function permissionWrapper(requiredPermission) {
+export function permissionWrapper(requiredPermission, allowInstanceAdmin = true) {
+  const acceptedPermissions = [requiredPermission];
+  if (allowInstanceAdmin) acceptedPermissions.push(SYSTEM_INSTANCE_ADMIN);
+
   return (request, response, next) => {
     const grantedPermissions = get(request, 'user.permissions') || [];
 
     logger.debug('Auth: checking permissions');
-    logger.debug(`Auth: expected permission: ${requiredPermission} || ${SYSTEM_INSTANCE_ADMIN}`);
+    logger.debug(`Auth: accepted permission: ${acceptedPermissions}`);
     logger.debug(`Auth: granted user permissions: ${grantedPermissions}`);
 
-    if (grantedPermissions.includes(requiredPermission) || grantedPermissions.includes(SYSTEM_INSTANCE_ADMIN)) {
+    if (acceptedPermissions.some(acceptedPermission => grantedPermissions.includes(acceptedPermission))) {
       logger.debug('Auth: permission check: PASSED');
-      next();
-    } else {
-      logger.warn('Auth: permission check: FAILED');
-      response.status(401)
-        .send({ message: `User missing expected permission: ${requiredPermission} || ${SYSTEM_INSTANCE_ADMIN}` })
-        .end();
+      return next();
     }
+    logger.warn('Auth: permission check: FAILED');
+    return response.status(401)
+      .send({ message: `User missing acceptable permission: ${acceptedPermissions}` });
   };
 }
 
 export function systemAdminPermissionWrapper() {
-  return (request, response, next) => {
-    const grantedPermissions = get(request, 'user.permissions') || [];
-
-    logger.debug('Auth: checking permissions');
-    logger.debug(`Auth: expected permission: ${SYSTEM_INSTANCE_ADMIN}`);
-    logger.debug(`Auth: granted user permissions: ${grantedPermissions}`);
+  return permissionWrapper(SYSTEM_INSTANCE_ADMIN, false);
+}
```

Comment:
Is there a reason to get rid of the logger statements?

Reply:
All of the same logging is carried out in `permissionWrapper` so these are now redundant.